### ISSUE TITLE
(publishValue()..connect()).first crashes

### ISSUE THREAD

**Issue by fdietze** [ReactiveX/rxdart#518]
I tried in the repl: https://repl.it/repls/LeftWelltodoModularity

```dart
import 'package:rxdart/rxdart.dart';


void main() {
	final sub = PublishSubject();
  final obs = sub.publishValue()..connect();

  sub.add(1);

  Future.microtask(() {
    obs.first.then((x) => print("first: $x"));
  });
}
```

I expected it to print:

```dart
first: 1
```

Error:

```
Unhandled exception:
NoSuchMethodError: The method 'cancel' was called on null.
Receiver: null
Tried calling: cancel()
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      _cancelAndValue (dart:async/stream_pipe.dart:59:35)
#2      Stream.first.<anonymous closure> (dart:async/stream.dart:1249:11)
#3      _RootZone.runUnaryGuarded (dart:async/zone.dart:1314:10)
#4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#5      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#6      _SyncBroadcastStreamController._sendData (dart:async/broadcast_stream_controller.dart:375:20)
#7      _BroadcastStreamController.add (dart:async/broadcast_stream_controller.dart:250:5)
#8      _StartWithStreamSink._safeAddFirstEvent (package:rxdart/src/transformers/start_with.dart:56:12)
#9      _StartWithStreamSink.onListen (package:rxdart/src/transformers/start_with.dart:37:11)
#10     forwardStream.<anonymous closure>.<anonymous closure> (package:rxdart/src/utils/forwarding_stream.dart:31:37)
#11     forwardStream.runCatching (package:rxdart/src/utils/forwarding_stream.dart:24:12)
#12     forwardStream.<anonymous closure> (package:rxdart/src/utils/forwarding_stream.dart:31:16)
#13     _runGuarded (dart:async/stream_controller.dart:805:24)
#14     _BroadcastStreamController._subscribe (dart:async/broadcast_stream_controller.dart:213:7)
#15     _ControllerStream._createSubscription (dart:async/stream_controller.dart:818:19)
#16     _StreamImpl.listen (dart:async/stream_impl.dart:472:9)
#17     DeferStream.listen (package:rxdart/src/streams/defer.dart:37:18)
#18     StreamView.listen (dart:async/stream.dart:1841:20)
#19     StreamView.listen (dart:async/stream.dart:1841:20)
#20     Stream.first (dart:async/stream.dart:1247:25)
#21     main.<anonymous closure> (file:///home/runner/LeftWelltodoModularity/main.dart:11:9)
#22     new Future.microtask.<anonymous closure> (dart:async/future.dart:202:37)
#23     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#24     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#25     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#26     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

One can work around the problem by giving a seed via `publishValueSeeded()`.

**Comment by hoc081098** [comment]
We fixed it in version '0.25.0-beta' or '0.25.0-beta2'. Please checkout it
https://repl.it/@hoc081098/LeftWelltodoModularity

**Comment by fdietze** [comment]
Looks good, thank you!
